### PR TITLE
CircleCI branch filtering on push-ci branch

### PR DESCRIPTION
**Pull request by ydshieh**
To avoid the branch `push-ci` (the branch to launch the actual Push CI on GitHub actions) to run `CircleCI` tests, as those are already run when the commits are merged into `main`.

The `filters` could not be added directly in the job definitions, like
```
    run_tests_torch_and_tf_all:
        filters:
            branches:
                ignore:
                    - circleci_branch_filtering
        working_directory: ~/transformers
```
nor
```
    run_tests_torch_and_tf_all:
        <<: *job_filters
        working_directory: ~/transformers
```
otherwise I got error
```
... [#/jobs/check_code_quality] extraneous key [filters] is not permitted|   |   Permitted keys:
```

See
https://stackoverflow.com/a/57366603

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Will defer to @LysandreJik but looks good to me!

**Review by LysandreJik — APPROVED** [review]
Looks good to me!

**Comment by ydshieh** [comment]
I will close this PR, once #17692 is merged (as it is preferred).

**Pull request closed without merging**
